src/reactionsystem.jl

```diff
@@ -134,36 +137,32 @@ function use_rate(kl::Num, react::Union{Vector{Num},Nothing}, stoich::Union{Vect
 end
 
 """ Get reagents """
-function getreagents(stoich::Dict{String,<:Real}, model::SBML.Model; rev=false)
-    if rev
-        stoich = Dict(k => -v for (k,v) in stoich)
-    end
-    reactants = Num[]
-    products = Num[]
-    rstoich = Float64[]
-    pstoich = Float64[]
-    for (k,v) in stoich
-        if v < 0
-            push!(reactants, create_var(k,Catalyst.DEFAULT_IV))
-            push!(rstoich, -v)
-            if model.species[k].boundary_condition == true
-                push!(products, create_var(k,Catalyst.DEFAULT_IV))
-                push!(pstoich, -v)
-            end
-        elseif v > 0
-            if model.species[k].boundary_condition != true
-                push!(products, create_var(k,Catalyst.DEFAULT_IV))
-                push!(pstoich, v)
```

Comment:
@paulflang okay yeah I think i see. what is the reason we don't push to reactants if bc is true here

Reply:
i guess because that would double things up right

Reply:
if bc is true we need to make sure that the stoichiometry of the species in question is the same on the right and left side of the reaction equation. however, we never want to modify the left hand side. this is because the left hand side can be used to determine if a reaction is mass action kinetics (i.e. A + B -> C gives mass action kinetic k*A*B, A -> B gives mass action kinetic k*A, etc.). This leaves us with the option to modify the product side. But this is already taken care of in line 149. In line 154 we only need to take care of the default case of bc = false, which is that reaction products are actually produced.